### ISSUE TITLE
Custom protobuf deserialization that works locally (VS Code) does not work in the cloud.

### ISSUE THREAD

**Issue by manucohenyashar** [Azure/azure-stream-analytics#89]
I created a custom serializer for deserializing protobuf messages into Azure Stream Analytics (ASA) input. I followed the tutorial in the ASA documentation and used the code in the ASA GitHub repo. I unit tested the deserializer, and then I ran a streaming job locally. Everything just works. So now it’s time to publish the job to Azure and test the job in a real environment. Unfortunately, in Azure, the situation is different. The same deserializer that works locally hangs in Azure.  The experience I see is that no data comes in the input. No errors, no logs, nothing. It's difficult to debug as in the Azure portal, there is no way to configure custom serialization, so it is difficult to see where the input takes the custom code from. I can see that the input is configured (in gray) to look for my custom deserializer in my .net assembly, but I cannot know where does it expect the assembly to be. Anyway, no data comes in. No errors, no logs, nothing. Pure fun. I attached the code and tests of the deserializer.

`public class BotStateDeserializer2 : StreamDeserializer<BotState>
    {
        StreamingDiagnostics diagnostics;

        public override IEnumerable<BotState> Deserialize(Stream stream)
        {
            if (stream == null)
            {
                diagnostics.WriteError("input stream is null", string.Empty);
            }

            while (stream.Position < stream.Length)
            {
                BotState item = null;
                try
                {
                    item = BotState.Parser.ParseDelimitedFrom(stream);
                }
                catch (Exception)
                {
                    try
                    {
                        stream.Position = 0;
                        item = BotState.Parser.ParseFrom(stream);
                    }
                    catch (Exception ex)
                    {
                        diagnostics.WriteError(ex.Message, ex.ToString());
                        throw;
                    }
                }

                yield return item;
            }
        }

        public override void Initialize(StreamingContext streamingContext)
        {
            diagnostics = streamingContext.Diagnostics;
        }
    }`

**Comment by sidramadoss** [comment]
@manucohenyashar - did you use [live testing](https://docs.microsoft.com/en-us/azure/stream-analytics/stream-analytics-live-data-local-testing) in Visual Studio to test your job with data flowing in directly from input sources in the cloud? If that works, you can deploy to the cloud and run the job. it should work and produce results (similar to the local live data testing in VS). Before you run the job, you can also [configure diagnostic logs ](https://docs.microsoft.com/en-us/azure/stream-analytics/stream-analytics-job-diagnostic-logs#send-diagnostics-to-azure-monitor-logs)on the job so that you can see any error your job runs into.

Unfortunately, you cant test your query (that using custom deserializers or C# UDFs) on the azure portal. That is why the recommended path is the test your job with live data from the cloud and then deploy/run it in the cloud. Let me know if this helps.

**Comment by manucohenyashar** [comment]
Thanks for your reply.

Yes, I did use live testing with data from a real event hub to test my job
in VS code, I did publish the job from VS code.
When running the job locally (with live data) it all worked great. After I
published the job to the cloud the same code did not work for the exact
same event hub messages.

Attached is the code that contains the serializer and deserializer.

Thanks

Manu
